### PR TITLE
[Scalability] rename to performance/scheduler

### LINKED ISSUE #1912: Scalability tests for Kueue scheduler
**What would you like to be added**:

Scalability tests (involving thousands of workloads) for Kueue scheduler, oriented on the performance of the Kueue algorithms, using the integration test layer to focus on the performance of the scheduler algorithms (for preemption and scheduling).

**Why is this needed**:

It can be handy to verify there is no regression when working on Fair sharing or Hierarchical cohorts features.

**Completion requirements**:

This enhancement requires the following artifacts:

- [x] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Move `test/performance` to `test/performance/e2e` and  `test/scalability` to `test/performance/scheduler`.

#### Which issue(s) this PR fixes:

Fixes #1912

#### Special notes for your reviewer:

The `run-scalability` make target will be removed after we merge this and update the CI config.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Read the [overview](https://kueue.sigs.k8s.io/docs/overview/) to learn more.
     [1.27](https://testgrid.k8s.io/sig-scheduling#periodic-kueue-test-e2e-main-1-27),
     [1.28](https://testgrid.k8s.io/sig-scheduling#periodic-kueue-test-e2e-main-1-28)
     on Kind.
-- ✔️ Scalability verification via [performance tests](https://github.com/kubernetes-sigs/kueue/tree/main/test/performance).
+- ✔️ Scalability verification via [performance tests](https://github.com/kubernetes-sigs/kueue/tree/main/test/performance/e2e).
```
Comment (alculquicondor) [inline README.md:44]: I would just leave the original link, so people can see both approaches.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

File: README.md (hunk 1 above)
Comment (trasc) [reply README.md]: done

**Pull request merged**